### PR TITLE
Fixed issue when including RmlUi/Core.h

### PR DESCRIPTION
**Pull request by larrymason01**
Let me know if anything I did could be changed, I'm sure some of the changes I made were unneeded

### CODE REVIEW

File: 3rdparty/CMakeLists.txt [hunk 1]
```diff
@@ -147,6 +147,6 @@ INSTALL(TARGETS freetype
         RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
        )
 
-SET(RML_SKIP_INSTALL ON CACHE BOOL "" FORCE)
+#SET(RML_SKIP_INSTALL ON CACHE BOOL "" FORCE)
 SET(CUSTOM_CONFIGURATION ON CACHE BOOL "" FORCE)
```
Comment (jonesmz) [inline 3rdparty/CMakeLists.txt:151]: don't comment out, just delete unused lines.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -23,6 +23,8 @@
 # SOFTWARE.
 ##
 
+SET(RMLUI_SKIP_INSTALL ON)
```
Comment (jonesmz) [inline src/CMakeLists.txt:26]: why did you move this here? It should go before the rmlui code is included, in the file where you commented it out.

Comment (larrymason01) [reply src/CMakeLists.txt:26]: I was unsure whether to keep it there or move it, thanks for the clarification.

Comment (jonesmz) [reply src/CMakeLists.txt:26]: In CMake, once you've created a "target", the only way to modify that target is through directly setting it's "target properties"

So here, setting a "loose" variable like this has no effect on anything.

That's why i said that you should be setting this variable in the 3rdparty/CMakeLists.txt file.

File: src/test_application/TestSystemInterface.h [hunk 3]
```diff
@@ -0,0 +1,5 @@
+#include "../../3rdparty/RmlUi/Include/RmlUi/Core.h"    //TODO: Still cannot include RmlUi in new files
+
+class TestSystemInterface : Rml::SystemInterface {
+
+};
```
Comment (Capital-Asterisk) [inline src/test_application/TestSystemInterface.h:5]: There should always be a (blank) newline at the end of text files. Some programs may not consider this a valid line and ignore it.

POSIX standards (practically anything non-windows) say that every line should be terminated by a newline character.

Unix text editors will add the blank line automatically, but apparently in VS there's a hidden setting: https://stackoverflow.com/questions/52894653/visual-studio-2017-insert-final-newline

Comment (larrymason01) [reply src/test_application/TestSystemInterface.h:5]: Noted, thanks for the feedback

**Comment by jonesmz** [comment]
Unfortunately we can't merge this PR until it passes the existing continuous integration.

Please try to get your change working in all the different platforms.

File: 3rdparty/CMakeLists.txt [hunk 4]
```diff
@@ -147,6 +147,6 @@ INSTALL(TARGETS freetype
         RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
        )
 
-SET(RML_SKIP_INSTALL ON CACHE BOOL "" FORCE)
```
Comment (jonesmz) [inline 3rdparty/CMakeLists.txt:150]: Why did you remove the CACHE BOOL "" FORCE?

File: src/CMakeLists.txt [hunk 5]
```diff
@@ -96,6 +96,8 @@ target_link_libraries(osp-magnum-deps PUBLIC
     toml11
     spdlog
     longeron
+    RmlCore
+    freetype
```
Comment (jonesmz) [inline src/CMakeLists.txt:100]: I don't believe that you need to explicitly link to freetype. RmlCore already does that for you.

**Comment by jonesmz** [comment]
@larrymason01 do you think you'll be able to complete this in the next couple of weeks?

**Comment by jonesmz** [comment]
> @larrymason01 do you think you'll be able to complete this in the next couple of weeks?

@larrymason01 Have you abandoned this PR? Please let me know one way or another.

**Comment by larrymason01** [comment]
Hey sorry! School got busy but I'm slowing down a bit, I'll be sure to get to it

**Comment by jonesmz** [comment]
> Hey sorry! School got busy but I'm slowing down a bit, I'll be sure to get to it

Still busy? :-)

**Comment by jonesmz** [comment]
@larrymason01 It's been several months since I asked last.

Closing.

Feel free to re-open in the future.
